contracts/contracts/gateway/GatewayMessengerFacet.sol

```diff
@@ -75,23 +91,9 @@ contract GatewayMessengerFacet is GatewayActorModifiers {
     }
 
     /**
-     * @dev propagates the populated cross net message for the given cid
-     * @param msgCid - the cid of the cross-net message
+     * @dev Propagates all the populated cross-net messages from the postbox.
      */
-    function propagate(bytes32 msgCid) external payable {
-        if (!s.multiLevelCrossMsg) {
-            revert MethodNotAllowed(ERR_MULTILEVEL_CROSS_MSG_DISABLED);
-        }
-
-        IpcEnvelope storage crossMsg = s.postbox[msgCid];
-
-        bool shouldBurn = LibGateway.commitCrossMessage(crossMsg);
-        // We must delete the message first to prevent potential re-entrancies,
-        // and as the message is deleted and we don't have a reference to the object
-        // anymore, we need to pull the data from the message to trigger the side-effects.
-        uint256 v = crossMsg.value;
-        delete s.postbox[msgCid];
-
-        LibGateway.crossMsgSideEffects({v: v, shouldBurn: shouldBurn});
+    function propagateAllPostboxMessages() external payable {
```

Comment:
Do we need to leak the "postbox" concept through the method name here? Why not just `propagateAll`?

Reply:
I think it makes sense to call it propagateAll here. But on the gateway interface it is then a bit vague. That's why I made it so specific. But I can revert it back.

Reply:
Maybe `propagateAllMessages`?